src/generate/generate.ts

```diff
@@ -1,17 +1,31 @@
-import { SigningAlgorithmConfig } from '../algorithms';
+import { SIGNING_ALGORITHM_CONFIG } from '../algorithms';
+import { normalizeHex } from '../utils/hex';
 export type PublicKey = {
-  key: string;
-  algorithm: SigningAlgorithmConfig;
+  key: string; // hex string
+  algorithm: keyof typeof SIGNING_ALGORITHM_CONFIG;
 };
 
 export function generate(...publicKeys: PublicKey[]): string {
   const pubKeys: object[] = publicKeys.map((publicKey, index) => {
+    if (
+      !Object.prototype.hasOwnProperty.call(
+        SIGNING_ALGORITHM_CONFIG,
+        publicKey.algorithm
```

Comment:
might be cleaner to map the values and then filter it

Reply:
When reading the code, something like `Object.values(SIGNING_ALGORITHM_CONFIG).map` is easier to parse